services/repository/src/main/java/org/sagebionetworks/file/controller/UploadController.java

```diff
@@ -360,6 +361,35 @@
 		// Pass it along
 		return fileService.createExternalS3FileHandle(userId, fileHandle);
 	}
+
+	/**
+	 * Create an GoogleCloudFileHandle to represent a Google Cloud Blob in a user's Google Cloud bucket.
+	 * <p>
+	 * In order to use this method an ExternalGoogleCloudStorageLocationSetting must
+	 * first be created for the user's bucket. The ID of the resulting
+	 * ExternalGoogleCloudStorageLocationSetting must be set in the
+	 * GoogleCloudFileHandle.storageLocationId. Only the user that created to the
+	 * ExternalGoogleCloudStorageLocationSetting will be allowed to create GoogleCloudFileHandle
+	 * using that storageLocationId.
+	 * </p>
+	 *
+	 * @param userId
+	 * @param fileHandle
+	 *            The GoogleCloudFileHandle to create
+	 * @return
+	 * @throws DatastoreException
+	 * @throws NotFoundException
+	 */
+	@ResponseStatus(HttpStatus.OK)
+	@RequestMapping(value = "/externalFileHandle/googleCloud", method = RequestMethod.POST)
```

Comment:
the constant "/externalFileHandle/googleCloud" should be defined in a file (URLHelpers.java??)